Add check for docker executable existence to Makefile

```
# make clean
make: docker: Command not found
make: *** [Makefile:61: image_clean] Error 127
```

**Step to reproduce:**
Exec `make clean` inside environment without `docker` executable (OpenPerf docker build image for example).

--- a/Makefile
+++ b/Makefile
@@ -58,7 +58,10 @@ image_pack: image
 
 .PHONY: image_clean
 image_clean:
-	@docker image rm -f ${DOCKER_IMAGE}; docker image prune -f; true
+ifneq ($(shell which docker), )
+	@docker image rm -f ${DOCKER_IMAGE}
+	@docker image prune -f
+endif
 
 # Generate code from schema definitions
 # Since generated code is stored in the repo, manual generation is only required

--- a/Makefile
+++ b/Makefile
@@ -58,8 +58,7 @@ image_pack: image
 
 .PHONY: image_clean
 image_clean:
-	@docker image rm -f ${DOCKER_IMAGE}
-	@docker image prune -f
+	@docker image rm -f ${DOCKER_IMAGE}; docker image prune -f; true
 
 # Generate code from schema definitions
 # Since generated code is stored in the repo, manual generation is only required

